sql/insert.go

```diff
@@ -98,9 +98,11 @@ func (p *planner) Insert(n *parser.Insert) (planNode, error) {
 
 	marshalled := make([]interface{}, len(cols))
 
-	var b client.Batch
+	b := client.Batch{}
+	result := &valuesNode{}
 	for rows.Next() {
 		rowVals := rows.Values()
+		result.rows = append(result.rows, rowVals)
```

Comment:
I think you should append `parser.DTuple{}` instead of `rowVals`. Appending `rowVals` is a bug waiting to happen because the return value from `row.Values()` is only valid until the next call to `rows.Next()`. Might even be better to just keep a count of the rows and then below do something like:

```
  return &valuesNode{rows: make([]parser.DTuple, count)}
```